src/main/java/com/wolApi/wolApi/api/bibleVerses/BibleVerseViews.java

```diff
@@ -21,20 +21,23 @@ public class BibleVerseViews {
     public BibleVerseViews(BibleVerseService bibleVerseService) {
         this.bibleVerseService = bibleVerseService;
     }
-    @GetMapping("/get-verse/{book}/{chapter}/{verse}")
-    public Verse getVerse(@PathVariable("book") String book,
-                          @PathVariable("chapter") String chapter,
-                          @PathVariable("verse") String verse) throws IOException {
+    @GetMapping("/get-verse/{bookNum}/{chapterNum}/{verseNum}")
+    public Verse getVerse(@PathVariable("bookNum") String bookNum,
+                          @PathVariable("chapterNum") String chapterNum,
+                          @PathVariable("verseNum") String verseNum) throws IOException, InterruptedException {
         return new Verse(
-                bibleVerseService.getVerse(book, chapter, verse)
+                bibleVerseService.getVerse(bookNum, chapterNum, verseNum),
+                bibleVerseService.getNumVersesInChapter(chapterNum),
+                bibleVerseService.getNumChaptersInBook(),
+                bibleVerseService.getBookName()
         );
     }
 
     @GetMapping("/get-verse/{book}/{chapter}")
-    public VerseList getVerses(@PathVariable("book") String book,
-                               @PathVariable("chapter") String chapter) throws IOException {
+    public VerseList getVerses(@PathVariable("bookNum") String bookNum,
+                               @PathVariable("chapterNum") String chapterNum) throws IOException {
```

Comment:
Also, the `GetMapping` args need to match the arguments. You changed to `bookNum` etc. so those changes need to be reflected in the `GetMapping`/

Reply:
Oh i thought i did that... ah well. That's probably why it was failing

Reply:
Btw they are reflected.... I think you just got confused with the diff :)

Reply:
No I didn't get confused

`@GetMapping("/get-verse/{book}/{chapter}")`
the `book`, `chapter` should in fact be `bookNum`, `chapterNum`, they are not reflected.

Reply:
Line 33 on that annotation

Reply:
oh yeah my bad I was looking at the PathVariable annotation (ง︡'-'︠)ง